internal/tiltfile/flags/args_def.go

```diff
@@ -0,0 +1,115 @@
+package flags
+
+import (
+	"bytes"
+	"errors"
+	"flag"
+	"fmt"
+
+	"go.starlark.net/starlark"
+
+	"github.com/windmilleng/tilt/internal/tiltfile/starkit"
+)
+
+type argValue interface {
+	flag() flag.Value
+	starlark() starlark.Value
+	setFromArgs([]string)
+}
+
+type argDef struct {
+	argValue
+	usage string
+}
+
+type ArgsDef struct {
+	positionalArgName string
+	args              map[string]argDef
+}
+
+func (ad ArgsDef) parse(args []string) (v starlark.Value, output string, err error) {
+	fs := flag.NewFlagSet("", flag.ContinueOnError)
+	w := &bytes.Buffer{}
+	fs.SetOutput(w)
+
+	for name, def := range ad.args {
+		if name == ad.positionalArgName {
+			continue
+		}
+		v := def.flag()
+		fs.Var(v, name, def.usage)
+	}
+
+	err = fs.Parse(args)
+	if err != nil {
+		return starlark.None, w.String(), err
+	}
+
+	if len(fs.Args()) > 0 {
+		if ad.positionalArgName == "" {
+			return starlark.None, w.String(), errors.New("positional args were specified, but none were expected (no arg defined with args=True)")
+		} else {
+			ad.args[ad.positionalArgName].setFromArgs(fs.Args())
+		}
+	}
+
+	ret := starlark.NewDict(len(ad.args))
+	for name, def := range ad.args {
+		err := ret.SetKey(starlark.String(name), def.starlark())
+		if err != nil {
+			return starlark.None, w.String(), err
+		}
+	}
+
+	return ret, w.String(), nil
+}
+
+// makes a new builtin with the given argValue constructor
```

Comment:
can you add some comments on why this is parameterized?

the extra function closure makes the code a lot harder to understand, and i don't understand why you did it this way or what value this adds

Reply:
I've expanded the comment, but I'm not sure if it helps much.
I suspect the main reason the value isn't apparent is that there's currently only one arg type, and once we add more arg types, it will be clearer.